src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Lock/GetAzureResourceLockCmdlet.cs

```diff
@@ -57,20 +64,51 @@ protected override void OnProcessRecord()
                 getNextPage: nextLink => this.GetNextLink<JObject>(nextLink),
                 cancellationToken: this.CancellationToken,
                 action: resources => this.WriteObject(sendToPipeline: this.GetOutputObjects(resources), enumerateCollection: true));
+
+            if (this.errors.Count != 0)
+            {
+                foreach (var error in this.errors)
+                {
+                    this.WriteError(error);
+                }
+            }
         }
 
         /// <summary>
         /// Queries the ARM cache and returns the cached resource that match the query specified.
         /// </summary>
         private async Task<ResponseWithContinuation<JObject[]>> GetResources()
         {
-            if (!string.IsNullOrWhiteSpace(this.LockName))
+            var response = new ResponseWithContinuation<JObject[]> { Value = new JObject[0] };
+
+            try
+            {
+                if (!string.IsNullOrWhiteSpace(this.LockName))
+                {
+                    var resource = await this.GetResource().ConfigureAwait(continueOnCapturedContext: false);
+
+                    response.Value = resource.AsArray();
+                }
+                else
+                {
+                    response = await this.ListResourcesTypeCollection().ConfigureAwait(continueOnCapturedContext: false);
+                }
+            }
+            catch (Exception ex)
             {
-                var resource = await this.GetResource().ConfigureAwait(continueOnCapturedContext: false);
-                return new ResponseWithContinuation<JObject[]> { Value = resource.AsArray() };
+                if (ex.IsFatal())
+                {
+                    throw;
+                }
+
+                ex = (ex is AggregateException)
+                    ? (ex as AggregateException).Flatten()
+                    : ex;
+
+                this.errors.Add(new ErrorRecord(ex, "ErrorGettingResourceLock", ErrorCategory.CloseError, null));
```

Comment:
@TianoMS can you please confirm if we do this for our other get cmdlets (rg, resources, policy, deployment, deployment operation, provider, feature)?